Core/src/IfcGeometryConverter/CurveConverter.h

```diff
@@ -2008,36 +2010,222 @@ namespace OpenInfraPlatform {
+				{
+					carve::geom::vector<3> centerToTrimPoint = trimPoint - ellipseCenter;
+					
+					if ((centerToTrimPoint.x / ellipseRadiusX) <= 1. && (centerToTrimPoint.y / ellipseRadiusY) <= 1.) {
+						double cosAngle = centerToTrimPoint.x / ellipseRadiusX;
+
+						if (abs(cosAngle) < 0.0001) {
+							if (centerToTrimPoint.y > 0.) {
+								return M_PI_2;
 							}
-							else if (center_trim_point.y < 0) {
-								result_angle = M_PI * 1.5;
+							else if (centerToTrimPoint.y < 0.) {
+								return 3 * M_PI_2;
 							}
 						}
 						else {
-							if (center_trim_point.y > 0) {
-								result_angle = acos(cos_angle);
+							if (centerToTrimPoint.y > 0.) {
+								return acos(cosAngle);
 							}
-							else if (center_trim_point.y < 0) {
-								result_angle = 2.0*M_PI - acos(cos_angle);
+							else if (centerToTrimPoint.y < 0.) {
+								return 2.0 * M_PI - acos(cosAngle);
+							}
+							else {
+								if (centerToTrimPoint.x > 0.) {
+									return 0.;
+								}
+								else {
+									return M_PI;
+								}
 							}
 						}
 					}
-					return result_angle;
+					else {
+						throw oip::InconsistentModellingException("The point is located outside the ellipse");
+					}
+				}
+
+				/**********************************************************************************************/
+				/*! \brief Calculates a trimming point on the Curve.
+				* \param[in] circle					A pointer to data from \c IfcCircle.
+				* \param[in] trimming				A pointer to data form \c IfcTrimmingSelect.
+				* \return							The location of the trimming point.
+				*/
+				carve::geom::vector<3> getPointOnCurve(const EXPRESSReference<typename IfcEntityTypesT::IfcCircle>& circle,
+					const typename IfcEntityTypesT::IfcTrimmingSelect & trimming) const throw(...)
+				{
+					switch (trimming.which())
+					{
+					case 0:
+					{
+						// Calculate a trimming point using \c IfcCartesianPoint. 
+						return getPointOnCurve(circle, trimming.get<0>());
+					}
+					case 1:
+					{
+						// Calculate a trimming point using \c IfcParameterValue.
+						return getPointOnCurve(circle, trimming.get<1>());
+					}
+					default:
+						throw oip::InconsistentGeometryException(circle, "TrimmingSelect is wrong!");
+					}
+				}
+
+				/**********************************************************************************************/
+				/*! \brief Calculates a trimming point on the curve using \c IfcCartesianPoint. .
+				* \param[in] circle					A pointer to data from \c IfcCircle.
+				* \param[in] cartesianPoint			A pointer to data form \c IfcCartesianPoint.
+				* \return							The location of the trimming point.
+				*/
+				carve::geom::vector<3> getPointOnCurve(const EXPRESSReference<typename IfcEntityTypesT::IfcCircle>& circle,
+					const EXPRESSReference<typename IfcEntityTypesT::IfcCartesianPoint>& cartesianPoint) const throw(...)
+				{
+
+					return placementConverter->convertIfcCartesianPoint(cartesianPoint);
+				}

```

Comment:
Please account for `circle->Position` as well (similarly to the other overload).

Reply:
Resolved: ad49034